jazzy branch possibly broken

Hi!

First of all, slightly off-topic, it would be nice to have at least 1 test launching the system. Right now only compilation is tested and issues with vendor and not vendor packages are not. (from what i can understand)

My current issue trying to rebase on vrx/jazzy is that [](http://packages.osrfoundation.org/gazebo/ubuntu-stable) provides sdformat `14.8.0` and the installed vendor package comes with `14.7.0`.

I see [https://github.com/gazebo-release/sdformat_vendor/pull/15](https://github.com/gazebo-release/sdformat_vendor/pull/15) , which is not merged. This means that when trying to launch the sw stack i get incompatible symbols.

Summary:
- gz-harmonic shipped with (http://packages.osrfoundation.org/gazebo/ubuntu-stable) depends on sdformat, and installs python bindings for `14.8.0` .
- ros-jazzy-desktop-full now installs vendor at `14.7.0` via:
    ```
    ros-jazzy-desktop-full 
      └── ros-jazzy-ros-gz-sim-demos 
          └── ros-jazzy-ros-gz-sim 
              └── ros-jazzy-sdformat-vendor
    ```

There are [many deps on vendor on the ros-gz codebase](https://github.com/gazebosim/ros_gz/blob/5995c031fe06b75ad3a1304c7a88323609b5b119/ros_gz_sim/package.xml#L30-L33)... so I don't thinkg building from source on a custom branch is a valid solution here.

Thoughts? @caguero @j-rivero

I'm experiencing a similar issue. It was working fine until last month, but after a fresh installation, it's no longer launching.

Here's the sequence of events:

1. Completed installation of ROS2 Jazzy + Gazebo Harmonic.
2. Installed osrf/vrx.
3. Upon launch, the following messages are displayed:

```
    [INFO] [launch]: All log files can be found below /home/username/.ros/log/2025-07-11-13-42-19-684968-hj-960XFH-89013
    [INFO] [launch]: Default logging verbosity is set to INFO
    [ERROR] [launch]: Caught exception in launch (see debug for traceback): "package 'ros_gz_sim' not found, searching: ['/home/username/vrx_ws/install', '/opt/ros/jazzy']"
```

 4. After installing ros-jazzy-ros-gz-sim using the `sudo apt-get install ros-jazzy-ros-gz-sim` command, I tried to launch again, and the following messages appeared:

```
    [INFO] [launch]: All log files can be found below /home/username/.ros/log/2025-07-11-13-43-57-237217-hj-960XFH-90145
    [INFO] [launch]: Default logging verbosity is set to INFO
    [ERROR] [launch]: Caught exception in launch (see debug for traceback): Caught multiple exceptions when trying to load file of format [py]:
    - ImportError: /usr/lib/python3/dist-packages/sdformat14.cpython-312-x86_64-linux-gnu.so: undefined symbol: _ZNK3sdf3v144Cone14AxisAlignedBoxEv
    - InvalidFrontendLaunchFileError: The launch file may have a syntax error, or its format is unknown
```

Please help. Is there a solution to this?

--- a/docker/Dockerfile.base
+++ b/docker/Dockerfile.base
@@ -1,5 +1,5 @@
-# Ubuntu 24.04 
-FROM osrf/ros:jazzy-desktop-full AS vrx-base
+# Ubuntu 24.04
+FROM ros:jazzy-ros-base AS vrx-base
 
 # Workaround from https://discourse.ros.org/t/ros-signing-key-migration-guide/43937
 RUN rm -f /etc/apt/sources.list.d/ros2-latest.list \
@@ -31,9 +31,51 @@ RUN echo $TZ > /etc/timezone && \
 RUN curl -s https://packages.osrfoundation.org/gazebo.gpg -o /usr/share/keyrings/pkgs-osrf-archive-keyring.gpg && \
     echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/pkgs-osrf-archive-keyring.gpg] http://packages.osrfoundation.org/gazebo/ubuntu-stable $(lsb_release -cs) main" | tee /etc/apt/sources.list.d/gazebo-stable.list > /dev/null
 
-# Install some 'standard' ROS packages and utilities.
+# Install gz-harmonic
 RUN apt update \
     && apt install -y --no-install-recommends \
        gz-harmonic \
+       equivs \
     && rm -rf /var/lib/apt/lists/* \
     && apt clean -qq
+
+# Solution for problem https://github.com/osrf/vrx/issues/839
+# Create fake providers for the ROS Gazebo vendor packages of the Gz libraries
+# that support Python bindings and use a colcon workspace to inject
+# the real ROS packages.
+
+# Create the dummy package control file for ROS Jazzy Gazebo vendor packages
+COPY <<EOF /tmp/fake-ros-jazzy-gz-vendor
+Package: fake-ros-jazzy-gz-vendor
+Version: 999.999
+Architecture: all
+Description: Fake ROS Jazzy Gazebo vendor packages for containers
+ This package provides dummy implementations to satisfy
+ ROS Jazzy Gazebo vendor package dependencies without actual functionality.
+Provides: ros-jazzy-gz-math-vendor, ros-jazzy-gz-msgs-vendor, ros-jazzy-gz-transport-vendor, ros-jazzy-gz-sim-vendor, ros-jazzy-sdformat-vendor
+EOF
+
+# Build and install the dummy package
+RUN cd /tmp \
+    && cat fake-ros-jazzy-gz-vendor \
+    && equivs-build fake-ros-jazzy-gz-vendor \
+    && dpkg -i fake-ros-jazzy-gz-vendor_*_all.deb \
+    && rm -f /tmp/fake-ros-jazzy-gz-vendor /tmp/fake-ros-jazzy-gz-vendor_*_all.deb
+
+ADD gz.repos /tmp/gz.repos
+
+RUN mkdir -p /opt/ros_gz_ws/src \
+    && cd /opt/ros_gz_ws \
+    && vcs import src < /tmp/gz.repos \
+    && apt update \
+    && rosdep install -r --from-paths . --ignore-src --rosdistro jazzy -y --skip-keys="gz_math_vendor gz_msgs_vendor gz_sim_vendor gz_transport_vendor sdformat_vendor" \
+    && . /opt/ros/jazzy/setup.sh \
+    && colcon build --merge-install --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
+
+# Source for interactive shells (both login and non-login)
+RUN echo "source /opt/ros_gz_ws/install/setup.bash" > /etc/profile.d/ros_gz_pythons.sh \
+    && echo "export GZ_CONFIG_PATH=\$GZ_CONFIG_PATH:/usr/share/gz/" > /etc/profile.d/ros_gz_pythons.sh \
+    && chmod +x /etc/profile.d/ros_gz_pythons.sh \
+    && echo "source /etc/profile.d/ros_gz_pythons.sh" >> /etc/bash.bashrc
+# Source for non-interactive shells
+ENV BASH_ENV=/etc/profile.d/ros_gz_pythons.sh

--- a/docker/gz.repos
+++ b/docker/gz.repos
@@ -1,0 +1,21 @@
+repositories:
+  gz_libs/gz_math_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_math_vendor.git
+    version: jazzy
+  gz_libs/gz_msgs_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_msgs_vendor.git
+    version: jazzy
+  gz_libs/gz_sim_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_sim_vendor.git
+    version: jazzy
+  gz_libs/gz_transport_vendor:
+    type: git
+    url: https://github.com/gazebo-release/gz_transport_vendor.git
+    version: jazzy
+  gz_libs/sdformat_vendor:
+    type: git
+    url: https://github.com/gazebo-release/sdformat_vendor.git
+    version: jazzy

